### PR TITLE
[YUNIKORN-2718] Assert invalid user name in Get User REST API

### PR DESCRIPTION
**Pull request by SophieTech88**
### What is this PR for?
Assert invalid user name in Get User REST API. 
In the pkg/webservice/handler.go file, we see in the func getUserResourceUsage, it didn't check whether the userName is valid or not. So in this PR, it check the UserRegExp to see if the user name is valid or not, and send back the InvalidUserName response.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2718

### How should this be tested?
It passed all test cases locally.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/webservice/handlers.go [hunk 1]
```diff
@@ -1100,6 +1101,10 @@ func getUserResourceUsage(w http.ResponseWriter, r *http.Request) {
 		buildJSONErrorResponse(w, err.Error(), http.StatusBadRequest)
 		return
 	}
+	if !configs.UserRegExp.MatchString(unescapedUser) {
+		buildJSONErrorResponse(w, InvalidUserName, http.StatusNotFound)
```
Comment (chenyulin0719) [inline pkg/webservice/handlers.go:1105]: IMO, if we call the User REST with an invalid username, the return code should be "400 Bad Request".   

@manirajv06 WDYT?

Comment (chia7712) [reply pkg/webservice/handlers.go:1105]: +1 to 400

Comment (SophieTech88) [reply pkg/webservice/handlers.go:1105]: When I convert to `400 Bad Request` code, the error message as below:
```
=== Deadlock detection enabled (timeout: 10 seconds, exit on deadlock: true, locking order disabled: false) ===
2024-07-15T22:07:44.687-0500    ERROR   core.config     configs/config.go:179   queue configuration validation failed   {"error": "undefined policy: invalid"}
github.com/apache/yunikorn-core/pkg/common/configs.ParseAndValidateConfig
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/common/configs/config.go:179
github.com/apache/yunikorn-core/pkg/common/configs.LoadSchedulerConfigFromByteArray
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/common/configs/config.go:152
github.com/apache/yunikorn-core/pkg/webservice.validateConf
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/webservice/handlers.go:152
github.com/apache/yunikorn-core/pkg/webservice.TestValidateConf
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/webservice/handlers_test.go:346
testing.tRunner
        /usr/local/go/src/testing/testing.go:1689
2024-07-15T22:07:44.688-0500    ERROR   core.config     configs/config.go:179   queue configuration validation failed   {"error": "MaxResources should be greater than zero in '' limit"}
github.com/apache/yunikorn-core/pkg/common/configs.ParseAndValidateConfig
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/common/configs/config.go:179
github.com/apache/yunikorn-core/pkg/common/configs.LoadSchedulerConfigFromByteArray
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/common/configs/config.go:152
github.com/apache/yunikorn-core/pkg/webservice.validateConf
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/webservice/handlers.go:152
github.com/apache/yunikorn-core/pkg/webservice.TestUserGroupLimits
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/webservice/handlers_test.go:387
testing.tRunner
        /usr/local/go/src/testing/testing.go:1689
2024-07-15T22:07:44.688-0500    ERROR   core.config     configs/config.go:179   queue configuration validation failed   {"error": "invalid resource combination for limit  all resource limits are null"}
github.com/apache/yunikorn-core/pkg/common/configs.ParseAndValidateConfig
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/common/configs/config.go:179
github.com/apache/yunikorn-core/pkg/common/configs.LoadSchedulerConfigFromByteArray
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/common/configs/config.go:152
github.com/apache/yunikorn-core/pkg/webservice.validateConf
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/webservice/handlers.go:152
github.com/apache/yunikorn-core/pkg/webservice.TestUserGroupLimits
        /Users/yunsun/Desktop/OpenSource/YUNIKORN/yunikorn-core/pkg/webservice/handlers_test.go:387
testing.tRunner
        /usr/local/go/src/testing/testing.go:1689
  ```

**Review by chenyulin0719 — APPROVED** [review]
+1 LGTM.  REST returned as expected.

@SophieTech88  Thanks for the update. Please create another PR to update error responses in doc page:

- https://yunikorn.apache.org/docs/next/api/scheduler#userstracker

image

**Comment by SophieTech88** [comment]
@chenyulin0719 Thank you so much. It is so exciting to see the  REST returned as expected. I will create another PR for doc.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**